src/parser.ts

```diff
@@ -68,44 +88,44 @@ function parseNonLiteral(
 ) {
 
   log(whiteBright.bgBlue('parser'), schema, '<-' + typeOfSchema(schema), processed.has(schema) ? '(FROM CACHE)' : '')
-
+  let scopedMakeName = () => { return makeName(options, schema, usedNames, keyName, keyNameFromDefinition) }
```

Comment:
Can you avoid the refactor here, and just add your new logic to the existing `standaloneName` function (modifying `standaloneName` to also take `options`)?